### PR TITLE
Make displaylist being sorted

### PR DESCRIPTION
**Pull request by feliwir**
I already did this change a while ago, but forgot to do a PR. The issue here was that objects in apt get rendered / handled in the wrong order since i though Dictionaries are sorted by default. Thankfully @Gwanw pointed that issue out to me.

### CODE REVIEW

File: src/OpenSage.Game/Gui/Apt/SpriteItem.cs [hunk 1]
```diff
@@ -345,7 +346,7 @@ public override void RunActions(TimeInterval gt)
 
         public override bool HandleInput(Point2D mousePos, bool mouseDown)
         {
-            foreach (var item in Content.Items.Values)
+            foreach (var item in Content.Items.Values.Reverse())
```
Comment (feliwir) [inline src/OpenSage.Game/Gui/Apt/SpriteItem.cs]: Buttons most be handled from top to bottom objects (in terms of depth)

File: src/OpenSage.Game/Gui/Apt/DisplayList.cs [hunk 2]
```diff
@@ -4,11 +4,11 @@ namespace OpenSage.Gui.Apt
 {
     public sealed class DisplayList
     {
-        public Dictionary<int, DisplayItem> Items { get; }
+        public SortedDictionary<int, DisplayItem> Items { get; }
```
Comment (feliwir) [inline src/OpenSage.Game/Gui/Apt/DisplayList.cs]: DisplayItems are sorted by a depth value. Before they were just randomly in here (depending on the insertion order)

File: src/OpenSage.Game/Gui/Apt/SpriteItem.cs (hunk 1 above)
Comment (tgjones) [reply src/OpenSage.Game/Gui/Apt/SpriteItem.cs]: It would be better to store these in the correct order rather than reversing the order every time?

Comment (feliwir) [reply src/OpenSage.Game/Gui/Apt/SpriteItem.cs]: The order is correct. The rendering happens exactly the opposite way, from far to near and happens much more frequently than the input handling :)

Comment (tgjones) [reply src/OpenSage.Game/Gui/Apt/SpriteItem.cs]: And then can you store both directions, so it doesn’t need to be computed every frame?

Comment (feliwir) [reply src/OpenSage.Game/Gui/Apt/SpriteItem.cs]: Yep, i’ll do that tomorrow

Comment (Tarcontar) [reply src/OpenSage.Game/Gui/Apt/SpriteItem.cs]: Fixed

**Pull request merged**